app/controllers/api/v1/rentals_controller.rb

```diff
@@ -0,0 +1,30 @@
+class RentalsController < ActionController::Base
+  def create
+    @rental = Rental.new(rental_params)
+    @station = Station.find(@rental[:station_id])
+    @rental.station = @station
+    @user = User.find(params[:user_id])
+    @rental.user = @user
+    unless @rental.save
+      render_error
+    end
+  end
+
+  def update
+    @rental = Rental.find(params[:id])
+    unless @rental.update(rental_params)
+      render_error
+    end
+  end
+
+  private
+
+  def render_error
+    render json: { errors: @rental.errors.full_messages },
+      status: :unprocessable_entity
+  end
+
+  def rental_params
+    params.require(:rental).permit(:in_progress, :station_id, :user_id)
```

Comment:
do we need the in_progress here? because it is true by default @The-Pavel  @rosenbelle @wanderlustwendy, what do you think?

Reply:
that was the most direct way we could think of to make an 'updated_at' timestamp. so that created_at timestamp becomes the starting time of charging and then when in_progress is changed to 'false' (meaning user charging time ended or user stopped charging early on the app), the 'updated_at' timestamp becomes ending time of charging. What do you guys think?